packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1227,6 +1270,112 @@ class RenderEditable extends RenderBox {
     }
   }
 
+  /// Sets the screen position of the floating cursor and the text position
+  /// closest to the cursor.
+  void setFloatingCursor(FloatingCursorDragState state, Offset boundedOffset, TextPosition lastTextPosition, { double resetLerpValue }) {
+    assert(state != null);
+    assert(boundedOffset != null);
+    assert(lastTextPosition != null);
+    if (state == FloatingCursorDragState.Start) {
+      _relativeOrigin = const Offset(0, 0);
+      _previousOffset = null;
+      _resetOriginOnBottom = false;
+      _resetOriginOnTop = false;
+      _resetOriginOnRight = false;
+      _resetOriginOnBottom = false;
+    }
+    _floatingCursorOn = state != FloatingCursorDragState.End;
+    _resetLerpValue = resetLerpValue;
+    if(_floatingCursorOn) {
+      _floatingCursorOffset = boundedOffset;
+      _floatingCursorTextPosition = lastTextPosition;
+    }
+    markNeedsPaint();
+  }
+
+  void _paintFloatingCaret(Canvas canvas, Offset effectiveOffset) {
+    assert(_textLayoutLastWidth == constraints.maxWidth);
+    final Paint paint = Paint()
+      ..color = _cursorColor;
+
+    double sizeAdjustmentX = _kFloatingCaretSizeOffset.dx;
+    double sizeAdjustmentY = _kFloatingCaretSizeOffset.dy;
+
+    if(_resetLerpValue != null) {
```

Comment:
Maybe resetAnimationValue? Also since this is in RenderEditable (which does a lot more other stuff), work the word floatingCursor somewhere into the name.